### DISCUSSION TITLE
Accounting API

### DISCUSSION THREAD

**Discussion by AlexanderTarler** [DevSkillsHQ/discusssions#24] [Help] [answered]
Hi there!

I'm currently working on the "Transaction Management Fullstack - Level 2"-test, but I'm having trouble reaching the Accounting API. I have tried using Insomnia, going to the url directly and making a request with my own code but I always get a 404. 

The instruction "The backend should implement the "Accounting API" I hopefully interpret correctly as making calls to that adress. I have tried every way to do it, even shortened the paths (down to accounting/get or accounting/get/ping) for example. Making requests to the url as a whole does nothing, not even /ping works. 

Am I wrong in thinking I should be making requests? Am I supposed to build the API myself? 

I hope it's alright to ask these questions here :)

**Comment by fenske** [comment, marked as answer]
Hi Alexander. 👋 I'm Anton from Alva. Happy to help!

The Accounting API is just a reference the candidate is expected to implement by building their own backend.

It's tricky with Fullstack assignments to not ask for "too much" so to speak. In this assignment, we e.g. tried to keep the balance by asking the candidate to only implement an in-memory backend storage to make the scope smaller yet allow the candidate to demonstrate their understanding of the backend.

Out of curiousity, would you say asking to "proxy" requests to a hosted API would be a better alternative?

**Comment by AlexanderTarler** [comment]
Thank you for the clarification! Much appreciated, Anton! :D 

In regards to your question -> I'm not sure, actually. If said API would be used for many other assignments, maybe? If it's only for this one though I'd probably keep it as is.

**Comment by jorgeuos** [comment]
Hi,

I understood the assignment as mentioned by @fenske . However if you want to try out the API with Insomnia I managed to do just that with this API endpoint: `https://infra.devskills.app/api/accounting/ping`

Example:
```sh
curl https://infra.devskills.app/api/accounting/ping
```

Br, Jorge

**Reply by fenske** [reply]
> Hi,
> 
> I understood the assignment as mentioned by @fenske . However if you want to try out the API with Insomnia I managed to do just that with this API endpoint: `https://infra.devskills.app/api/accounting/ping`
> 
> Example:
> 
> ```shell
> curl https://infra.devskills.app/api/accounting/ping
> ```
> 
> Br, Jorge

Hehe. That's indeed a working API (e.g. it's used in a Transaction Management Frontend challenge where the candidate is asked to integrate against it).

Not that long ago it was also linked in the API reference for the Fullstack challenge (to give a chance to play with it) but we removed any mention of it as some candidates got confused assuming they were supposed to "wrap" it instead of building their own backend.

**Reply by jorgeuos** [reply]
Ah, I see, I thought I read it somewhere.
